cohttp-eio/src/io.ml

```diff
@@ -0,0 +1,52 @@
+module IO = struct
+  type 'a t = 'a
+
+  let ( >>= ) v f = f v
+  let return v = v
+
+  type ic = Eio.Buf_read.t
+  type oc = Eio.Buf_write.t
+  type conn = Eio.Net.stream_socket
+
+  let refill ic =
+    try
+      let () = Eio.Buf_read.(ensure ic (buffered_bytes ic + 1)) in
+      `Ok
+    with End_of_file -> `Eof
+
+  let with_input_buffer ic ~f =
+    let contents = Eio.Buf_read.peek ic in
+    let res, consumed =
+      f (Cstruct.to_string contents) ~pos:0 ~len:(Cstruct.length contents)
+    in
+    let () = Eio.Buf_read.consume ic consumed in
+    res
+
+  let read_line ic =
```

Comment:
Could this just be `Eio.Buf_read.line`? The difference seems to be that this one discards the final line if it doesn't have a line terminator, whereas Eio's one returns it, but I'm not sure if that's intentional.

Reply:
~~My understanding is that `Eio.Buf_read.line` also stops on `\n`, which we do not want here.~~ We actually do want to stop on `\n` because RFC7230 states parser MAY accept `\n` as terminator. The more you know 🌈